9 post /api/articles/article id/comments

Cralus opened the pull request:
Implemented the post comment functionality on the /api/articles/:article_id/comments endpoint with associated error hadnling for bad requests

Cralus committed:
completed the initial happy path for posting a new comment

Cralus committed:
implemented error handling for adding a comment by article id

daviidbartlett commented on __tests__/app.test.js:
```diff
@@ -258,6 +258,81 @@ describe('GET /api/articles/:article_id/comments', () => {
+            expect(msg).toBe("Bad Request");
+          });
+    })
+    test('should respond with a status code of 400 when given a article_id that doesnt exist', () => {
```
I'd say this is a 404, article not found

daviidbartlett commented on __tests__/app.test.js:
```diff
@@ -258,6 +258,81 @@ describe('GET /api/articles/:article_id/comments', () => {
+        return request(app)
+        .post('/api/articles/1/comments')
+        .expect(400)
+        .send({ username: 'someone', body: 'Thought provoking read!' })
```
again, 404

daviidbartlett commented on __tests__/app.test.js:
```diff
@@ -258,6 +258,81 @@ describe('GET /api/articles/:article_id/comments', () => {
+            expect(msg).toBe("Bad Request");
+          });
+    })
+    test('should respond with a status code of 400 when given a object with too few properties', () => {
```
nice idea. I think a better description would be "missing *required* fields" because you may find that in the future some made be optional and depending how you handle this in your code it might not scale very well

daviidbartlett commented on models/comments.js:
```diff
@@ -18,3 +18,20 @@ exports.fetchCommentsByArticleId = (articleId) => {
+exports.addCommentByArticleId = (comment, articleId) => {
+    if(Object.keys(comment).length !== 2)
+    {
+        return Promise.reject({ status: 400, msg: "Bad Request" });
```
yeah this won't scale well if the comments table undergoes migrations. (see previous comment on test)

Cralus committed:
improved the tests for too few items to be more robust and changed the username not existing error and the article not existing error to 404 Not Found

Cralus commented:
Implemented all the comments and removed the test for too many fields on the set object as only the required ones are accessed, so extras should not be an issue and the size of the sent object might increase in future.
